pkg/templateservicebroker/servicebroker/bind.go

```diff
@@ -219,32 +181,55 @@ func (b *Broker) Bind(u user.Info, instanceID, bindingID string, breq *api.BindR
 
 	credentials := map[string]interface{}{}
 
-	resp := b.updateCredentials(u, namespace, instanceID, kapi.GroupName, "configmaps", credentials, func(o metav1.ListOptions) (runtime.Object, error) {
-		return b.extkc.Core().ConfigMaps(namespace).List(o)
-	})
-	if resp != nil {
-		return resp
-	}
+	for _, object := range templateInstance.Status.Objects {
+		switch object.Ref.GroupVersionKind().GroupKind() {
+		case kapi.Kind("ConfigMap"),
+			kapi.Kind("Secret"),
+			kapi.Kind("Service"),
+			routeapi.Kind("Route"),
+			routeapi.LegacyKind("Route"):
```

Comment:
does this mean the expose annotation can only be used on these object types?

Reply:
Yes; that is currently the case anyway.

Reply:
Nothing in the code should prevent expansion in the future.

Reply:
hm, i guess that was a choice we made, sorry, i forgot about it.

Reply:
Why is legacy only an issue for Route?  aren't there also Legacy apis for the other types?

Reply:
Legacy is via /oapi, non-legacy is via /apis/<group>.  Core Kubernetes API objects haven't moved from /api into /apis/<something>, at least at this time.

Reply:
hm.  I thought it was also group vs non-group apis?

Reply:
i guess you're saying the k8s objects haven't been groupified.  ok.